booster_nrpe/booster_nrpe.py

```diff
@@ -154,16 +132,16 @@ def read(self, data):
             response = struct.unpack(">2hih1024s", data)
         except struct.error as err:  # bad format...
             self.rc = 3
-            self.message = "Error : cannot read output from nrpe daemon ; datalen=%s : err=%s" % (len(data), err)
-            return (self.rc, self.message)
-
-        self.rc = response[3]
-        # the output is padded with \x00 at the end so
-        # we remove it.
-        self.message = re.sub('\x00.*$', '', response[4])
-        crc_orig = response[2]
+            self.message = ("Error : cannot unpack output ; "
+                            "datalen=%s : err=%s" % (len(data), err))
+        else:
+            self.rc = response[3]
+            # the output is padded with \x00 at the end so
+            # we remove it.
+            self.message = re.sub('\x00.*$', '', response[4])
```

Comment:
after double think about it : if a plugin has responded to the nrpe server with such a message.. then I think it could also probably send that to us.. and effectively with the re.sub() that would make the entire message be stripped.. hmm.. don't know what to do... a plugin which respond with the 0x00 binary in front of what should be a (nearly dumb-)human readable message.. would be well.. I don't know ;)